rust/src/message/payload/mod.rs

```diff
@@ -0,0 +1,56 @@
+use std::{borrow::Borrow, ops::Range};
+
+pub(crate) mod sum;
+pub use self::sum::*;
+pub(crate) mod sum2;
+pub use self::sum2::*;
+pub(crate) mod update;
+pub use self::update::*;
+
+use derive_more::From;
+
+use crate::{
+    mask::{Mask, MaskedModel},
+    message::traits::ToBytes,
+    LocalSeedDict,
+};
+
+/// Payload of a [`Message`]
+#[derive(From, Eq, PartialEq, Clone, Debug)]
+pub enum Payload<D, M, N> {
+    /// Payload of a sum message
+    Sum(Sum),
+    /// Payload of an update message
+    Update(Update<D, M>),
+    /// Payload of a sum2 message
+    Sum2(Sum2<N>),
+}
+
+pub type PayloadOwned = Payload<LocalSeedDict, MaskedModel, Mask>;
+
+impl<D, M, N> ToBytes for Payload<D, M, N>
+where
+    D: Borrow<LocalSeedDict>,
+    M: Borrow<MaskedModel>,
+    N: Borrow<Mask>,
+{
+    fn buffer_length(&self) -> usize {
+        match self {
+            Payload::Sum(m) => m.buffer_length(),
+            Payload::Sum2(m) => m.buffer_length(),
+            Payload::Update(m) => m.buffer_length(),
+        }
+    }
+
+    fn to_bytes<T: AsMut<[u8]>>(&self, buffer: &mut T) {
+        match self {
+            Payload::Sum(m) => m.to_bytes(buffer),
+            Payload::Sum2(m) => m.to_bytes(buffer),
+            Payload::Update(m) => m.to_bytes(buffer),
+        }
+    }
+}
+
+pub(crate) const fn range(start: usize, length: usize) -> Range<usize> {
+    start..(start + length)
+}
```

Comment:
can we move this to `message/mod.rs` or to `utils.rs`? was quite hard to find.

Reply:
You mean `message/utils.rs` I guess? Yes totally. I should probably move stuff out of the `mod.rs`

Reply:
`message/utils.rs` would work well